pokete_classes/fightmap.py

```diff
@@ -197,226 +200,241 @@ def get_item(self, items, inv):
                 time.sleep(2)
-                self.outp.outp(f"{obj.ext_name} has used all its' attacks!")
+                self.outp.outp(f"{player.curr.ext_name} has used all its' attacks!")
                 time.sleep(3)
             if winner is not None:
-                if (obj.identifier != "__fallback__"
-                        and not winner.player
-                        and any(p.hp > 0 for p in figure.pokes[:6])
-                        and ask_bool(self,
-                                     "Do you want to choose another Pokete?")):
-                    old_player = player
-                    players, player = self.choose_poke(figure, players,
-                                                       player, enemy)
-                    if old_player == player:
+                loser = providers[(providers.index(winner) + 1) % 2]
+                if (
+                    loser.curr.player
+                    and any(p.hp > 0 for p in loser.pokes[:6])
+                    and ask_bool(self, "Do you want to choose another Pokete?")
+                ):
+                    success = self.choose_poke(loser)
+                    if not success:
                         break
+                elif (
+                    type(loser) is Trainer
+                    and winner.curr.player
+                    and any(p.hp > 0 for p in loser.pokes)
+                ):
+                    time.sleep(1)
+                    ico = loser.curr.ico
+                    self.fast_change([ico, self.deadico1, self.deadico2], ico)
+                    self.deadico2.remove()
+                    self.show()
+                    self.clean_up(loser)
+                    loser.play_index += 1
+                    self.add_1(*providers)
+                    ico = loser.curr.ico
+                    self.fast_change(
+                        [ico, self.deadico2, self.deadico1, ico], ico
+                    )
+                    self.outp.outp(f"{loser.name} used {loser.curr.name}!")
+                    self.show()
+                    time.sleep(2)
                 else:
                     break
-            obj = [i for i in players if i != obj][-1]
-            enem = [i for i in players if i != obj][-1]
-        loser = [obj for obj in players if obj != winner][0]
-        _xp = (loser.lose_xp + (1 if loser.lvl() > winner.lvl() else 0))\
-                             * (2 if info["type"] == "duel" else 1)
-        self.outp.outp(f"{winner.ext_name} won!" + (f'\nXP + {_xp}'
-                                                    if winner.player else ''))
-        if winner.player and info["type"] == "duel":
+            index += 1
+        # loser = [obj for obj in players if obj != winner][0]
+        _xp = sum(
+            [
+                poke.lose_xp + max(0, poke.lvl() - winner.curr.lvl())
+                for poke in loser.pokes
+            ]
+        ) * (2 if type(loser) is not NatureProvider else 1)
```

Comment:
Remove the `[]` around the list comprehension, `sum()` can operator on generators AFAIK